addons/mail/static/src/utils/test_utils.js

```diff
@@ -257,7 +254,99 @@ async function beforeEach(self) {
         }
         data[modelName] = { fields, records };
     }
+    return data;
+}
+
+let pyEnv;
+/**
+ * Creates an environment that can be used to setup test data as well as
+ * creating data after test start.
+ *
+ * @returns {Object} An environment that can be used to interact with the mock
+ * server (creation, deletion, update of records...)
+ */
+ export async function startServer() {
+    pyEnv = new Proxy(
+        {
+            mockServer: new MockServer(await getInitialServerData(), {}),
+            ...TEST_USER_IDS,
+        },
+        {
+            get(target, name) {
+                if (target[name]) {
+                    return target[name];
+                }
+                return {
+                    /**
+                     * Simulate a 'create' operation on a model.
+                     *
+                     * @param {Object[] | Object} values records to be created.
+                     * @returns {Number[] | Number} array of ids if more than one value was passed,
+                     * id of created record otherwise.
+                     */
+                    create(values = {}) {
+                        if (!Array.isArray(values)) {
+                            values = [values];
+                        }
+                        const recordIds = values.map(value => target.mockServer._mockCreate(name, value))
+                        return recordIds.length === 1 ? recordIds[0] : recordIds;
+                    },
+                    /**
+                     * Simulate a 'search' operation on a model.
+                     *
+                     * @param {Array} domain
+                     * @param {Object} context
+                     * @returns {Number[]} array of ids corresponding to the given domain.
+                     */
+                    search(domain, context = {}) {
+                        return target.mockServer._mockSearch(name, [domain], context);
+                    },
+                    /**
+                     * Simulate a 'search_read' operation on a model.
+                     *
+                     * @param {Array} domain
+                     * @param {Object} context
+                     * @returns {Object[]} array of records corresponding to the given domain.
+                     */
+                    searchRead(domain, context = {}) {
+                        return target.mockServer._mockSearchRead(name, [], { domain, context });
+                    },
+                    /**
+                     * Simulate an 'unlink' operation on a model.
+                     *
+                     * @param {Number[]} ids
+                     * @returns {true} mockServer 'unlink' method always returns true.
```

Comment:
nit-picking again, but `@returns {x}` the `x` is always a type, not a value. So here `{boolean}`.